Source/Scene/ModelExperimental/PointCloudStylingPipelineStage.js

```diff
@@ -233,11 +255,40 @@ const builtinVariableSubstitutionMap = {
   NORMAL: "attributes.normalMC",
 };
 
-const parameterList = "ProcessedAttributes attributes";
+function getVariableSubstitutionMap(propertyAttributes) {
+  const variableSubstitutionMap = clone(builtinVariableSubstitutionMap);
+
+  if (!defined(propertyAttributes)) {
+    return variableSubstitutionMap;
+  }
+
+  for (let i = 0; i < propertyAttributes.length; i++) {
+    const propertyAttribute = propertyAttributes[i];
+    const properties = propertyAttribute.properties;
+    for (const propertyId in properties) {
+      if (properties.hasOwnProperty(propertyId)) {
+        const variableName = sanitizeGlslIdentifier(propertyId);
+        variableSubstitutionMap[variableName] = `metadata.${variableName}`;
+      }
+    }
+  }
+
+  return variableSubstitutionMap;
+}
+
+function sanitizeGlslIdentifier(identifier) {
+  // For use in the shader, the property ID must be a valid GLSL identifier,
+  // so replace invalid characters with _
+  return identifier.replaceAll(/[^_a-zA-Z0-9]+/g, "_");
+}
```

Comment:
Hm, at this point we have (at least?) 3 places where we sanitize identifiers (here, the `MetadataPipelineStage`, and in `PntsLoader`). Could we move this to a function in `ModelExperimentalUtility`? If so, base the implementation on `PntsLoader`, which had a slightly more sophisticated function. Though the handling of empty string and duplicate values might be best to let the caller handle